### PR TITLE
Creation of restaurant and pole tables

### PR DESCRIPTION
**Pull request by Anthony-AGTN**

### CODE REVIEW

File: back-end/src/models/Restaurant/Restaurant.entity.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  @Column({ nullable: true })
+  @Field()
+  closeAt?: Date;
+
+  @ManyToOne(() => Pole, { eager: true, onDelete: "CASCADE" })
+  @Field(() => Pole, { nullable: false })
+  pole: Pole;
+
+  @Field(() => String)
```
Comment (vincentDubresson) [inline back-end/src/models/Restaurant/Restaurant.entity.ts:52]: Interest of getName() ?

File: back-end/src/models/Restaurant/Restaurant.entity.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  getName() {
```
Comment (vincentDubresson) [inline back-end/src/models/Restaurant/Restaurant.entity.ts:53]: Interest of getName()

File: back-end/src/resolvers/Pole/Pole.input.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+@ArgsType()
+class CreatePoleArgs {
+  @Field()
+  @MinLength(1, {
+    message: "Le nom doit faire au moins un caractère de long.",
+  })
+  name: string;
+
+  @Field()
```
Comment (vincentDubresson) [inline back-end/src/resolvers/Pole/Pole.input.ts:12]: l.12 - l.22 : Check validations

File: back-end/src/resolvers/Pole/Pole.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  @Mutation(() => Pole)
+  createPole(
+    @Args() { name, address, zipCode, city, email }: CreatePoleArgs
+  ): Promise<Pole> {
+    return PoleRepository.createPole(name, address, zipCode, city, email);
+  }
+
+  @Mutation(() => Pole)
+  updatePoleName(
```
Comment (vincentDubresson) [inline back-end/src/resolvers/Pole/Pole.resolver.ts:22]: updatePole instead of updatePoleName :)

**Review by vincentDubresson — CHANGES_REQUESTED** [review]

File: back-end/src/models/Restaurant/Restaurant.entity.ts (hunk 2 above)
Comment (Anthony-AGTN) [reply back-end/src/models/Restaurant/Restaurant.entity.ts:53]: Ok getName is delete

File: back-end/src/resolvers/Pole/Pole.input.ts (hunk 3 above)
Comment (Anthony-AGTN) [reply back-end/src/resolvers/Pole/Pole.input.ts:12]: Not time today

File: back-end/src/resolvers/Pole/Pole.resolver.ts (hunk 4 above)
Comment (Anthony-AGTN) [reply back-end/src/resolvers/Pole/Pole.resolver.ts:22]: Ok it is good

**Review by vincentDubresson — APPROVED** [review]

**Pull request merged**